drivers/ethernet/eth_sam_gmac_priv.h

```diff
@@ -31,7 +31,19 @@
 /** Memory alignment of the RX/TX Buffer Descriptor List */
 #define GMAC_DESC_ALIGNMENT               4
 /** Total number of queues supported by GMAC hardware module */
-#define GMAC_QUEUE_NO                     3
+#if defined(CONFIG_SOC_ATMEL_SAME70_REVB) || \
+	defined(CONFIG_SOC_ATMEL_SAMV71_REVB)
+#define GMAC_QUEUE_NUM                    6
+#elif !defined(CONFIG_SOC_SERIES_SAM4E)
+#define GMAC_QUEUE_NUM                    3
+#else
+#define GMAC_QUEUE_NUM                    1
+#endif
+#define GMAC_QUEUE_PRI_NUM                (GMAC_QUEUE_NUM - 1)
+#if (GMAC_QUEUE_NUM > 1)
+BUILD_ASSERT_MSG(ARRAY_SIZE(GMAC->GMAC_TBQBAPQ) + 1 == GMAC_QUEUE_NUM,
+		 "GMAC_QUEUE_NUM doesn't match soc header");
+#endif
 /** Number of priority queues used */
 #define GMAC_PRIORITY_QUEUE_NO            (CONFIG_ETH_SAM_GMAC_QUEUES - 1)
```

Comment:
We renamed `GMAC_QUEUE_NO` to `GMAC_QUEUE_NUM`. We probably should also rename `GMAC_PRIORITY_QUEUE_NO` to `GMAC_PRIORITY_QUEUE_NUM`.

Reply:
This naming scheme is quite confusing indeed.

`GMAC_QUEUE_NUM` and `GMAC_QUEUE_PRI_NUM` refer to "the number of (priority) queues supported in hardware", whereas `GMAC_PRIORITY_QUEUE_NO` refers to "the number of active/used (priority) queues."

I think we should just rename `GMAC_PRIORITY_QUEUE_NO` to something like `GMAC_ACTIVE_PRIORITY_QUEUE_NUM` and define `GMAC_ACTIVE_QUEUE_NUM` as `CONFIG_ETH_SAM_GMAC_QUEUES` and use that in place of it as well.

```
#define GMAC_QUEUE_NUM                    ...
#define GMAC_PRIORITY_QUEUE_NUM           (GMAC_QUEUE_NUM - 1)

#define GMAC_ACTIVE_QUEUE_NUM             CONFIG_ETH_SAM_GMAC_QUEUES 
#define GMAC_ACTIVE_PRIORITY_QUEUE_NUM    (GMAC_ACTIVE_QUEUE_NUM - 1)
```

Reply:
Used @stephanosio suggestion.